clang/lib/CodeGen/BackendUtil.cpp

```diff
@@ -990,8 +990,14 @@ void EmitAssemblyHelper::EmitAssembly(BackendAction Action,
   }
 
   // Allocate static local memory in SYCL kernel scope for each allocation call.
-  if (LangOpts.SYCLIsDevice)
+  if (LangOpts.SYCLIsDevice) {
+    // Group local memory pass depends on inlining. Turn it on even in case if
+    // all llvm passes or SYCL early optimizations are disabled.
+    // TODO: Remove this workaround when dependency on inlining is eliminated.
```

Comment:
```suggestion
    // FIXME: Remove this workaround when dependency on inlining is eliminated.
```
(Typically, TODO suggests it'll be done as part of the PR, FIXME suggests we'll get it fixed up later.)